oauthproxy.go

```diff
@@ -601,13 +601,12 @@ func (p *OAuthProxy) Authenticate(rw http.ResponseWriter, req *http.Request) int
 	}
 
 	if p.PassRolesHeader {
-		type RoleProvider interface {
-			GetUserRoles() string
+		switch rp := p.provider.(type) {
+		case providers.RoleProvider:
+			roles := rp.GetUserRoles()
+			log.Printf("User role data - %v", roles)
+			req.Header["X-Forwarded-Roles"] = []string{roles}
 		}
-		rp := p.provider.(RoleProvider);
-		rp.GetUserRoles();
-		log.Printf("User role data - %v", rp.GetUserRoles())
-		req.Header["X-Forwarded-Roles"] = []string{rp.GetUserRoles()}
 	}
```

Comment:
This check is unnecessary. If it somehow fails at runtime, it's a programmer error because it get's checked at startup so it's okay to let the builtin panic handle it. If it was necessary, it might be better to use `if rp, ok := p.provider.(providers.RoleProvider); ok { ...` since there's only one case.